client/src/main/java/io/pravega/client/stream/StreamCut.java

```diff
@@ -26,8 +25,24 @@
      * This is used represents an unbounded StreamCut. This is used when the user wants to refer to the current HEAD
      * of the stream or the current TAIL of the stream.
      */
-    StreamCut UNBOUNDED = new UnboundedStreamCut();
-
+    public static final StreamCut UNBOUNDED = new StreamCut() {
```

Comment:
This will cause the issue observed in https://github.com/pravega/pravega/issues/2502

Reply:
No, it just solves it in a different way. Instead of allowing multiple objects and implementing equals() to have them be considered the same. The method:
```
        private Object readResolve() {
            return UNBOUNDED;
        }
```
Ensures that only one instance ever exists. This means that equals() and == will both work. 

The reason for moving the object to an anonymous class was so that it can be a singleton. A separate class could not be because it's constructor would have to be visible.